src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,214 @@
+  protected isCorrected = !!this.getCorrectedFilingDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
+    return startDateMax
+  }
+
+  /** The maximum start date string. */
+  protected get startDateMaxStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMax)
+  }
+
+  /** The recognition date or business start date-time string. */
+  protected get recognitionDateTime (): string {
+    if (this.isBenIaCorrectionFiling) {
+      if (this.getCorrectedFilingDate) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectedFilingDate, true)
+      }
+    }
+    if (
+      this.isFirmCorrectionFiling ||
+      this.isAlterationFiling ||
+      this.isChangeRegFiling ||
+      this.isFirmConversionFiling
+    ) {
+      if (this.getCorrectedFilingDate) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectedFilingDate, true)
+      } else if (this.getBusinessFoundingDate) {
+        return this.apiToPacificDateLong(this.getBusinessFoundingDate)
+      }
+    }
+    return 'Unknown'
+  }
+
+  protected onChangeClicked (): void {
+    this.onEditMode = true
+  }
+
+  /** Called to add a new date. */
+  protected onOkClicked (date: string): void {
+    if (date) {
+      this.newCorrectedStartDate = date
+    }
+  }
+
+  protected onDoneClicked (): void {
+    if (this.newCorrectedStartDate) {
+      if (this.yyyyMmDdToPacificDate(this.newCorrectedStartDate, true) !==
+        this.apiToPacificDateLong(this.getBusinessFoundingDate)) {
+        this.setCorrectionFilingDate(this.newCorrectedStartDate)
+        this.isCorrected = true
+      } else {
+        this.newCorrectedStartDate = undefined
+      }
+    }
+    this.onEditMode = false
+  }
+
+  protected onCancelClicked (): void {
+    this.onEditMode = false
+  }
+
+  protected onUndoClicked (): void {
+    this.setCorrectionFilingDate(undefined)
+    this.newCorrectedStartDate = undefined
+    this.isCorrected = false
+  }
+
+  @Watch('onEditMode', { immediate: true })
+  protected syncValidity (): void {
+    const isValid = !this.onEditMode
+    this.setValidComponent({ key: 'isValidStartDate', value: isValid })
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+
+</style>
```

Comment:
You may need some styling in here to make the title the correct colour (gray9).

Reply:
From the UXpin in shows the title as being black colour #212925 the same as the label but I will verify that with Yui

Reply:
Title and label may mean the same thing.

I only noticed that Create UI is explicit about setting the label (left side), and I think the data (right side) is defaulted. They are typically a different gray value. If it works correctly then you're set (and probably don't need this style section).

Reply:
Ya the component in create UI is displayed as its own component where this is part of the larger YourCompany component I think that is where the difference comes from.

Reply:
OK. You can delete empty style blocks if you don't need them.